### ISSUE TITLE
feat(premium): Add ability to redeem a premium token

### ISSUE THREAD

**Issue by HenryLeC** [oasis-sh/oasis#333]

**Comment by bineetNaidu** [comment]
Could You add More description onto this :smiley:

**Comment by HenryLeC** [comment]
Yeah so we basically want a link with a uid such that when you click it you will redeem x amounts of premium. Like Discord Nitro

**Comment by bineetNaidu** [comment]
> Yeah so we basically want a link with a uid such that when you click it you will redeem x amounts of premium. Like Discord Nitro

So we need stripe integration. cool then give more info pricing or a charge for a redemption.  And i would like to implement this. :smile:

**Comment by HenryLeC** [comment]
> > Yeah so we basically want a link with a uid such that when you click it you will redeem x amounts of premium. Like Discord Nitro
> 
> 
> 
> So we need stripe integration. cool then give more info pricing or a charge for a redemption.  And i would like to implement this. :smile: 

For rn we don't need stripe. Now we just want to have the ability to redeem them.

**Comment by heybereket** [comment]
> > > Yeah so we basically want a link with a uid such that when you click it you will redeem x amounts of premium. Like Discord Nitro
> > 
> > 
> > So we need stripe integration. cool then give more info pricing or a charge for a redemption.  And i would like to implement this. 😄
> 
> For rn we don't need stripe. Now we just want to have the ability to redeem them.

@bineetNaidu I think Henry is referring to something like discord.gift/{code}, but I may be wrong.

**Comment by bineetNaidu** [comment]
@HenryLeC yep even i think the same thing but if you give a good description on how this will be modeled then it will be easier for me and we all will be on the same page. otherwise i have to guess and do the task.

**Comment by Angshu31** [comment]
@HenryLeC What can the user redeem from these premium tokens? OasisNitro?

**Comment by HenryLeC** [comment]
> @HenryLeC What can the user redeem from these premium tokens? OasisNitro?

We don't have a name for it yet, but yes it will be like OasisNitro

**Comment by f1shy-dev** [comment]
possible names:

Beach+
Relaxer
Fruity
Pro
Beach Pro
Oasis Pro
Beacher
Sunbather

idk beach/oasis related names that have a like good vibe to them would sound nice
